projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-get-expiring-notice-content.tsx

```diff
@@ -0,0 +1,218 @@
+import { Col, Text } from '@automattic/jetpack-components';
+import { gmdateI18n } from '@wordpress/date';
+import { __, sprintf } from '@wordpress/i18n';
+import { ProductsList } from './products-list';
+
+interface Props {
+	productSlug: string;
+	expiredAlertType: 'expired' | 'expiring-soon';
+	productName: string;
+	expiryDate: string;
+	productsEffected: string[];
+}
+
+export const useGetExpiringNoticeContent = ( {
+	productSlug,
+	expiredAlertType,
+	productName,
+	expiryDate,
+	productsEffected,
+}: Props ) => {
+	if ( ! productSlug ) {
+		return null;
+	}
+
+	// Remove the billing term suffix from the purchase product_slug. The notice will be the same regardless of the expiring product's billing term.
+	const product = productSlug.replace( /(?:_t1|_t2|_bi)?(?:_yearly|_monthly)/, '' );
+	const expiryDateFormatted = gmdateI18n( 'M j, Y', expiryDate );
+
+	const noticeHeading =
+		expiredAlertType === 'expired'
+			? sprintf(
+					// translators: %s is the Jetpack product name, i.e.- Jetpack Backup, Jetpack Security, etc..
+					__( 'Your %1$s plan has expired', 'jetpack-my-jetpack' ),
+					productName
+			  )
+			: sprintf(
+					// translators: %1$s is the Jetpack product name, i.e.- Jetpack Backup, Jetpack Security, etc..
+					__( 'Heads up! Your %1$s plan is about to expire', 'jetpack-my-jetpack' ),
+					productName
+			  );
+
+	// Notice Content
+	switch ( product ) {
```

Comment:
I feel like this switch will get bigger and bigger as we add more products with specific messaging. I was thinking if we could have the notice text content in one place and a somewhat generic function that uses that text content to build the JSX a bit more dynamically. I don't have a strong opinion right now, though. I'm mostly interested in having this code a bit easier to extend.

Reply:
Yeah I initially had the notice content split up into 2 files, one for expired products, and one for expiring soon, which simplified each file quite a bit, but then it was suggested that I combine the files into one, adding more conditionals and so forth possibly making things a bit more confusing to extend. But, given the nature of each product maybe needing different text, different number of paragraphs, different jsx structure, and possibly displaying other differences (such as icons, images, etc), I fear trying to combine and generalize/optimize any more on this file... For now anyway until we know more exactly what the content of each product will look like. We can always clean up or refactor this file more as needed as we add more content for each specific product. 👍😃